File: ui/src/services/api/worktreesService.ts
/**
 * Worktree API service
 */

import { apiPost, apiDelete } from './api-client.js';
import type { RepositoryData } from '../../types/domain.js';

interface CreateWorktreeResponse {
  taskId: string;
  [key: string]: unknown;
}

interface DeleteWorktreeResponse {
  data: RepositoryData;
}

/**
 * Create a new worktree
 */
export async function createWorktree(
  org: string,
  repo: string,
  branch: string | null = null,
  prompt: string | null = null
): Promise<{ taskId: string; data: unknown }> {
  const body: { org: string; repo: string; branch?: string; prompt?: string } = { org, repo };
  if (branch) {
    body.branch = branch;
  }
  if (prompt) {
    body.prompt = prompt;
  }

  const data = await apiPost<CreateWorktreeResponse>(
    '/api/worktrees',
    body,
    { errorPrefix: 'Failed to create worktree' }
  );

  return {
    taskId: data && typeof data.taskId === 'string' ? data.taskId : '',
    data: data,
  };
}

/**
 * Delete a worktree
 */
export async function deleteWorktree(org: string, repo: string, branch: string): Promise<RepositoryData> {
  const response = await apiDelete<DeleteWorktreeResponse>(
    '/api/worktrees',
    { org, repo, branch },
    { errorPrefix: 'Failed to delete worktree' }
  );
  return response.data || {};
}

File: mobile/ios/AgentrixMobile/Services/WorktreesService.swift
import Foundation

final class WorktreesService {
    private let api: AgentrixAPIClient

    init(api: AgentrixAPIClient) {
        self.api = api
    }

    struct CreateWorktreeResponse: Decodable {
        let taskId: String
        let org: String
        let repo: String
        let branch: String?
    }

    func createWorktree(org: String, repo: String, branch: String?, prompt: String?) async throws -> CreateWorktreeResponse {
        struct Payload: Encodable {
            let org: String
            let repo: String
            let branch: String?
            let prompt: String?
        }
        return try await api.request(
            "/api/worktrees",
            method: .post,
            body: Payload(org: org, repo: repo, branch: branch, prompt: prompt)
        )
    }

    func deleteWorktree(org: String, repo: String, branch: String) async throws {
        struct Payload: Encodable {
            let org: String
            let repo: String
            let branch: String
        }
        try await api.requestVoid(
            "/api/worktrees",
            method: .delete,
            body: Payload(org: org, repo: repo, branch: branch)
        )
    }
}
